lib/requester/requester.js

```diff
@@ -99,17 +101,34 @@ Requester.prototype.request = function (item, cb, scope) {
                 },
                 responseCookies: cookies
             },
+            contentEncoding = response.responseHeaders && response.responseHeaders['Content-Encoding'],
+            encodedData = contentEncoding && (rawResponse ? rawResponse : util.getEncodedBody(
+                response.responseBody, contentEncoding)),
             responseObj = { // @todo get rid of jsonifyResponse
                 code: response.responseCode.code,
                 header: response.responseHeaders,
                 body: responseString,
                 stream: resBody,
                 cookie: undefined, // @todo get from cookieJar and implement loading/unloading in SDK
-                responseTime: responseTime
+                responseTime: responseTime,
+                encodedData: encodedData
             };
 
         cb.call(scope || this, null, response, actualRequest, responseObj, request);
-    });
+    })
+    .on('response', function(response) {
+        if(response.headers['content-encoding']) {
+            var chunks = [];
+            response.on('data', function(chunk) {
+                chunks.push(chunk);
+            })
+            .on('end', function() {
+                chunks.forEach(function(chunk) {
+                    rawResponse += util.arrayBufferToString(chunk);
```

Comment:
Even then. Currently app is slow on large response only because of this reason - that we convert to string right away. Is it good to add one more thing that adds to this burden? Ideally we could expose only the raw size and later expose raw response buffer itself so that it can be consumed directly from the buffer in app and Newman. 

@czardoz thoughts?